File: clients/console/src/types/paper.ts
export interface Paper {
  id: string
  title: string
  authors: string
  source_url: string
  raw_text: string
  status: PaperStatus
  created_at: string
  updated_at: string
}

export type PaperStatus = 'uploaded' | 'analyzing' | 'analyzed' | 'reviewing' | 'merged'

export interface PaperCreateRequest {
  title: string
  authors: string
  source_url?: string
  raw_text: string
}

export interface PaperUpdateRequest {
  title?: string
  authors?: string
  source_url?: string
  status?: string
}

export interface PapersListResponse {
  papers: Paper[]
  total: number
}

export interface ExtractedNode {
  name: string
  description: string
  relevance: number
}

export interface ExtractedEdge {
  source: string
  target: string
  relation: string
  weight: number
}

export interface ReviewQueueItem {
  id: string
  paper_id: string
  extracted_nodes: ExtractedNode[]
  extracted_edges: ExtractedEdge[]
  status: ReviewStatus
  reviewer_id: string | null
  review_notes: string | null
  created_at: string
  reviewed_at: string | null
}

export type ReviewStatus = 'pending' | 'approved' | 'rejected'

export interface ReviewQueueListResponse {
  items: ReviewQueueItem[]
  total: number
}

File: services/console/internal/model/paper.go
package model

import (
	"encoding/json"
	"time"
)

// Paper represents a research paper (mirrors core model).
type Paper struct {
	ID        string    `json:"id"`
	Title     string    `json:"title"`
	Authors   string    `json:"authors"`
	SourceURL string    `json:"source_url"`
	RawText   string    `json:"raw_text"`
	HasPDF    bool      `json:"has_pdf"`
	Status    string    `json:"status"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

// PaperCreateRequest is the payload for creating a paper via core.
type PaperCreateRequest struct {
	Title     string `json:"title"`
	Authors   string `json:"authors"`
	SourceURL string `json:"source_url"`
	RawText   string `json:"raw_text"`
	PDFBase64 string  `json:"pdf_base64,omitempty"`
}

// PaperUpdateRequest is the payload for updating a paper via core.
type PaperUpdateRequest struct {
	Title     *string `json:"title,omitempty"`
	Authors   *string `json:"authors,omitempty"`
	SourceURL *string `json:"source_url,omitempty"`
	Status    *string `json:"status,omitempty"`
}

// PapersListResponse wraps a list of papers.
type PapersListResponse struct {
	Papers []Paper `json:"papers"`
	Total  int     `json:"total"`
}

// ReviewQueueItem represents a review queue entry (mirrors core model).
type ReviewQueueItem struct {
	ID             string          `json:"id"`
	PaperID        string          `json:"paper_id"`
	ExtractedNodes json.RawMessage `json:"extracted_nodes"`
	ExtractedEdges json.RawMessage `json:"extracted_edges"`
	Status         string          `json:"status"`
	ReviewerID     *string         `json:"reviewer_id"`
	ReviewNotes    *string         `json:"review_notes"`
	CreatedAt      time.Time       `json:"created_at"`
	ReviewedAt     *time.Time      `json:"reviewed_at"`
}

// ReviewQueueListResponse wraps a list of review items.
type ReviewQueueListResponse struct {
	Items []ReviewQueueItem `json:"items"`
	Total int               `json:"total"`
}
